build/building/cache/garbage-collection.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Garbage collection
+keywords: build, buildx, buildkit, garbage collection, prune
+---
+
+While [`docker build prune`](../../../engine/reference/commandline/builder_prune.md)
+or [`docker buildx prune`](../../../engine/reference/commandline/buildx_prune.md)
+commands run at once, garbage collection runs periodically and follows an
+ordered list of prune policies.
+
+## Configuration
+
+Depending on the [driver](../drivers/index.md) used by your builder instance,
+the garbage collection will use a different configuration file.
+
+If you're using the [`docker` driver](../drivers/docker.md), garbage collection
+can be configured in the [Docker Daemon configuration](../../../engine/reference/commandline/dockerd.md#daemon-configuration-file).
+file:
+
+```json
+{
+  "builder": {
+    "gc": {
+      "enabled": true,
+      "defaultKeepStorage": "10GB",
+      "policy": [
+          {"keepStorage": "10GB", "filter": ["unused-for=2200h"]},
+          {"keepStorage": "50GB", "filter": ["unused-for=3300h"]},
+          {"keepStorage": "100GB", "all": true}
+      ]
+    }
+  }
+}
+```
+
+For other drivers, garbage collection can be configured using the
+[BuildKit configuration](https://github.com/moby/buildkit/blob/master/docs/buildkitd.toml.md)
+file:
+
+```toml
+[worker.oci]
+  gc = true
+  gckeepstorage = 10000
+  [[worker.oci.gcpolicy]]
+    keepBytes = 512000000
+    keepDuration = 172800
+    filters = [ "type==source.local", "type==exec.cachemount", "type==source.git.checkout"]
+  [[worker.oci.gcpolicy]]
+    all = true
+    keepBytes = 1024000000
+```
+
+## Default policies
+
+Default garbage collection policies are applied to all builders if not
+already set:
+
+```
+GC Policy rule#0:
+        All:            false
+        Filters:        type==source.local,type==exec.cachemount,type==source.git.checkout
+        Keep Duration:  48h0m0s
+        Keep Bytes:     512MB
+GC Policy rule#1:
+        All:            false
+        Keep Duration:  1440h0m0s
+        Keep Bytes:     26GB
+GC Policy rule#2:
+        All:            false
+        Keep Bytes:     26GB
+GC Policy rule#3:
+        All:            true
+        Keep Bytes:     26GB
+```
```

Comment:
Where does this come from, it looks like command output? (partly for curiosity, but also we should include the command used so that users can check their own policies themselves).

Reply:
Default policies:
* buildkitd: https://github.com/moby/buildkit/blob/master/cmd/buildkitd/config/gcpolicy.go
* dockerd: https://github.com/moby/moby/blob/master/builder/builder-next/worker/gc.go

Reply:
Oh awesome, ok, let's not put that command in then, since it's a bit fiddly :smile:

Reply:
> Maybe we could add this in buildx when inspecting a builder? :thinking:

Yes! That would be really useful info to pull through :tada: